src/autoscaler/metricsforwarder/server/custom_metrics_handlers.go

```diff
@@ -0,0 +1,208 @@
+			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+				Code:    "Interal-Server-Error",
+				Message: "Error reading custom metrics request body"})
+			return
+		}
+		credentials = models.CustomMetricCredentials{
+			Username: usernameHash,
+			Password: passwordHash,
+		}
+		// update the cache
+		mh.credentialCache.Set(appID, credentials, mh.cacheTTL)
+
+		isValid = mh.validateCredentials(username, credentials.Username, password, credentials.Password)
+		// If Credentials in DB is not valid
+		if !isValid {
+			mh.logger.Error("error-validating-authorizaion-header", err)
+			handlers.WriteJSONResponse(w, http.StatusUnauthorized, models.ErrorResponse{
+				Code:    "Authorization-Failure-Error",
+				Message: "Basic authorization credential does not match"})
+			return
+		}
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		mh.logger.Error("error-reading-request-body", err, lager.Data{"body": r.Body})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Error reading custom metrics request body"})
+		return
+	}
+	var metricsConsumer *models.MetricsConsumer
+	err = json.Unmarshal(body, &metricsConsumer)
+	if err != nil {
+		mh.logger.Error("error-unmarshaling-metrics", err, lager.Data{"body": r.Body})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Error unmarshaling custom metrics request body"})
+		return
+	}
+	isValidRequest, err := mh.validateCustomMetricTypes(appID, metricsConsumer)
+	if !isValidRequest && err != nil {
+		mh.logger.Error("failed-validating-metrictypes", err, lager.Data{"metrics": metricsConsumer})
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad-Request",
+			Message: "Error validating custom metrics type"})
+		return
+	}
+
+	metrics := mh.getMetrics(appID, metricsConsumer)
+
+	if len(metrics) <= 0 {
+		mh.logger.Debug("failed-parsing-custom-metrics-request-body", lager.Data{"metrics": metrics})
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad-Request",
+			Message: "Error parsing request body"})
+		return
+	}
+
+	mh.logger.Debug("custom-metrics-parsed-successfully", lager.Data{"metrics": metrics})
+	for _, metric := range metrics {
+		mh.metricForwarder.EmitMetric(metric)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (mh *CustomMetricsHandler) validateCredentials(username string, usernameHash string, password string, passwordHash string) bool {
+	isUsernameAuthenticated := bcrypt.CompareHashAndPassword([]byte(usernameHash), []byte(username))
+	isPasswordAuthenticated := bcrypt.CompareHashAndPassword([]byte(passwordHash), []byte(password))
+	if isPasswordAuthenticated == nil && isUsernameAuthenticated == nil { // password matching successfull
```

Comment:
The variable name looks like a "bool" type , but it is not .. 
how about using 
usernameAuthErr := ...
passwordAuthErr := ...
if usernameAuthErr == nil && passwordAuthErr == nil 

or let us omit the separated variable here

Reply:
Renamed the variables